GoldRush.js

```diff
@@ -0,0 +1,125 @@
+class GoldRush extends Matrix{
+    constructor(rows, columns){
+        super()
+        this.matrix= []
+        //probabilities: s-stone, c-coin, n-nothing
+        this.distribution = {
+            S: 30,
+            N: 10,
+            C: 60
+        }
+        this.coinsSum = 0;
+        this.createBoard(rows, columns)
+        this.player1 = {x:0, y:0, score: 0, turn: true}
+        this.player2 = {x: columns - 1, y: rows - 1, score: 0, turn:true}
+    }
+    createBoard(columns, rows){
+        for(let i=0; i<rows; i++){
+            this.matrix[i] = []
+            for(let j=0; j< columns; j++){
+                this.matrix[i][j] = this.getRandomValue(i, j)
+            }
+            const isRowAllS = this.printRow(i).every(r => r==='S')
+            if(isRowAllS){
+                this.matrix[i][Math.floor(columns/2)]= 'C' 
+                this.coinsSum += 10
+            } 
+        }
+        let isColumnAllS
+        for(let i=0; i< this.matrix[0].length; i++){
+            isColumnAllS = this.printColumn(i).every(c => c === 'S')
+            if(isColumnAllS) this.matrix[i][Math.floor(rows/2)]
+        }
+        if(this.matrix[0][0] === 'C') this.coinsSum -= 10
+        this.matrix[0][0] = 1
+        if(this.matrix[rows-1][columns-1] === 'C') this.coinsSum -= 10 
+        this.matrix[rows-1][columns-1] = 2
+    }
+    movePlayer(player, direction){
+        let currentPlayer
+        if(player === 1){
+            currentPlayer = this.player1
+        }else{
+            currentPlayer = this.player2
+        }
+        if(currentPlayer.turn){
+            if(direction === 'left'){
+                if(currentPlayer.y !== 0 && 
+                    this.matrix[currentPlayer.x][currentPlayer.y - 1] !== 'S' &&
+                    !Number.isInteger(this.matrix[currentPlayer.x][currentPlayer.y - 1])){
+                    this.matrix[currentPlayer.x][currentPlayer.y] = '.'
+                    if(this.matrix[currentPlayer.x][currentPlayer.y - 1] === 'C'){
```

Comment:
You should try to make more helper methods instead of repeating yourself and making really long functions

so make an `isWall()` and `checkCoin()` function that will do that for you and make your `movePlayer()` function smaller